cmd/thanos/query.go

```diff
@@ -141,6 +141,19 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application) {
 
 		promql.SetDefaultEvaluationInterval(time.Duration(*defaultEvaluationInterval))
 
+		flagsMap := map[string]string{}
+
+		// Exclude kingpin default flags to expose only Thanos ones.
+		boilerplateFlags := kingpin.New("", "").Version("")
+
+		for _, f := range cmd.Model().Flags {
+			if boilerplateFlags.GetFlag(f.Name) != nil {
+				continue
+			}
+
+			flagsMap[f.Name] = f.Value.String()
```

Comment:
We should skip over setting this if `f.Value.String()` is empty otherwise we will get a bunch of options in the API response with keys but empty values.

Reply:
I was thinking that, but I thought it would still be helpful if someone wants to see that if a particular flag has been set or not. wdyt?

Reply:
Double checked and it seems like Prometheus does the same. I guess it looked a bit weirder on Thanos because there are more flags without any default values. But, you are correct, let's leave these for clarity.